crates/credentials/src/verifiable_credential.rs

```diff
@@ -206,64 +252,87 @@ impl VerifiableCredential {
+        if let Some(issuance_date) = &vc_payload.issuance_date {
+            if issuance_date != &nbf {
+                return Err(CredentialError::IssuanceDateMismatch);
+            }
         }
 
-        if vc.issuance_date == i64::default() {
-            vc.issuance_date = nbf;
-        } else if vc.issuance_date != nbf {
-            return Err(CredentialError::IssuanceDateMismatch);
+        if let Some(expiration_date) = &vc_payload.expiration_date {
+            if exp.is_none() {
+                return Err(CredentialError::MisconfiguredExpirationDate(
+                    "VC has expiration date but no exp in registered claims".to_string(),
+                ));
+            }
+
+            if expiration_date != &exp.unwrap() {
+                return Err(CredentialError::IssuanceDateMismatch);
+            }
         }
 
-        // if exp exists, make sure there is not a mismatch and assign it to vc expiration date
-        // if vc expiration date exists and exp does not, throw a misconfigured exp error
         if exp.is_some() {
-            if vc.expiration_date.is_some() {
-                if vc.expiration_date != exp {
-                    return Err(CredentialError::ExpirationDateMismatch);
-                }
-
-                let now = Utc::now().timestamp();
-                if now > exp.unwrap() {
-                    return Err(CredentialError::CredentialExpired(
-                        "The verifiable credential has expired".to_string(),
-                    ));
-                }
-
-                vc.expiration_date = exp;
+            let now = Utc::now().timestamp();
+            if now > exp.unwrap() {
+                return Err(CredentialError::CredentialExpired(
+                    "The verifiable credential has expired".to_string(),
+                ));
             }
-        } else if vc.expiration_date.is_some() {
-            return Err(CredentialError::MisconfiguredExpirationDate(
-                "VC has expiration date but no exp in registered claims".to_string(),
-            ));
         }
 
+        let mut vc_issuer = Issuer::String(iss.clone());
+
+        if let Some(Issuer::Object(_)) = vc_payload.issuer.as_ref() {
+            vc_issuer = vc_payload.issuer.clone().unwrap();
+        }
+
+        let vc_credential_subject =
+            vc_payload
+                .credential_subject
+                .clone()
+                .unwrap_or(CredentialSubject {
+                    id: sub,
+                    params: None,
+                });
+
+        let vc = VerifiableCredential {
+            context: vc_payload.context.clone(),
+            id: jti.clone(),
+            r#type: vc_payload.r#type.clone(),
+            issuer: vc_issuer.clone(),
+            issuance_date: nbf,
+            expiration_date: exp,
+            credential_subject: vc_credential_subject,
+        };
+
         validate_vc_data_model(&vc).map_err(CredentialError::ValidationError)?;
```

Comment:
I wonder if `validate_data_model` would be useful as a public method on `VerifiableCredential`

Reply:
Yup that's how we had it many moons ago, but decided it was too hard for the end user to do vc.verify() vs vc.validate()

so was going with that latest api "spec"

until we have a more concrete api spec that @KendallWeihe has been working on and / or we just really want to do it lol.